bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/AsyncCompletionProposalPopup.java

```diff
@@ -389,11 +412,46 @@ protected List<CompletableFuture<List<ICompletionProposal>>> buildCompletionFutu
 					return Collections.emptyList();
 				}
 				return proposals;
-			}));
+			}, EXECUTOR));
 		}
 		return futures;
 	}
 
+	/**
+	 * Submit a task in such a way that it actually reacts to cancellation (i.e. calls to
+	 * {@code future.cancel(true)})
+	 *
+	 * @param executor Do not use the common pool here since that one does not cancel (interrupts)
+	 *            worker threads
+	 * @return an interruptible future.
+	 */
+	private static <T> CompletableFuture<T> submitInterruptible(
```

Comment:
All this codes makes me wonder why not using a plain `FutureTask` in the first place?